src/unistd.rs

```diff
@@ -466,12 +473,39 @@ pub fn sethostname(name: &[u8]) -> Result<()> {
     Errno::result(res).map(drop)
 }
 
-pub fn gethostname(name: &mut [u8]) -> Result<()> {
-    let ptr = name.as_mut_ptr() as *mut c_char;
-    let len = name.len() as size_t;
+/// Get the host name and store it int the provided buffer, returning the actual
+/// length stored in the buffer (see
```

Comment:
Yeah, I need to update this doc.  When I was first writing this I though the system call returned the length but it does not.